### PR TITLE
UP-4830: Compile less from gradle

### PR DESCRIPTION
**Pull request by ChristianMurphy**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -411,6 +411,7 @@ modules.xml
 # Project specific ignores
 /build.properties
 /data/
+**/respondr/defaultSkin.css
```
Comment (drewwills) [inline .gitignore]: What's the point of this line?  The generated `defaultSkin.css` file will be in the `build/` directory, which is already ignored.

Comment (ChristianMurphy) [reply .gitignore]: I'm compiling the file into the source folder to avoid modifying the `war` step.

**Comment by drewwills** [comment]
```
"scripts": {
  "compile-less": "lessc src/main/webapp/media/skins/respondr/defaultSkin.less src/main/webapp/media/skins/respondr/defaultSkin.css"
}
```

This part needs to be more like...

```
"scripts": {
  "compile-less": "lessc src/main/webapp/media/skins/respondr/defaultSkin.less build/generated-sources/lessc/main/webapp/media/skins/respondr/defaultSkin.css"
}
```

And then we need to figure out how to add `build/generated-sources/lessc/main/webapp` to the war archive.

File: .gitignore (hunk 1 above)
Comment (drewwills) [reply .gitignore]: Please -- let's not start doing that kind of thing.

EVERY product of the build goes into `build/`.

**Comment by ChristianMurphy** [comment]
The `war` plugin expects a single source folder. So I modified it to copy the webapp folder to the build, the to compile to the less into build/webapp

**Comment by drewwills** [comment]
@ChristianMurphy -- I cherry-picked your commits, and they've been incorporated in the PR.

**Pull request closed without merging**